metricbeat/mb/parse/url.go

```diff
@@ -244,18 +245,22 @@ func getURL(
 			return nil, t, fmt.Errorf("error parsing URL: empty host")
 		}
 
-		// Validate the host. The port is optional.
-		host, _, err := net.SplitHostPort(u.Host)
+		// Validate the host.
+		host, port, err := net.SplitHostPort(u.Host)
 		if err != nil {
 			if strings.Contains(err.Error(), "missing port") {
 				host = u.Host
 			} else {
-				return nil, t, fmt.Errorf("error parsing URL: %v", err)
+				return nil, t, fmt.Errorf("error parsing URL: %w", err)
 			}
 		}
 		if host == "" {
 			return nil, t, fmt.Errorf("error parsing URL: empty host")
 		}
+		// Add default port to host if port is empty and defaultport is set
+		if port == "" && defaultPort != "" {
```

Comment:
is there a case where both are empty?

Reply:
Yes could happen. Eg: When URL (without port) is parsed and default port is not specified. Since this url parsing is used by many metricsets, the behaviour will stay as it is for them.

Reply:
what happens in that case? should default port always be specified?

Reply:
No, there can be cases where defaultport is not present and its ok.